lib/cookie.d.ts

```diff
@@ -108,98 +108,102 @@ export function permutePath (path: string): string[];
  * Base class for CookieJar stores. Available as tough.Store.
  */
 export class Store {
-  // TODO(blakeembrey): Finish this.
-  // https://github.com/SalesforceEng/tough-cookie#store
-}
-
-/**
- * A just-in-memory CookieJar synchronous store implementation, used by default.
- * Despite being a synchronous implementation, it's usable with both the
- * synchronous and asynchronous forms of the CookieJar API.
- */
-export class MemoryCookieStore extends Store {}
-
-/**
- * Interface for the options of the Cookie constructor
- */
-export interface CookieConstructorOptions {
-  /**
-   * the name or key of the cookie (default "")
-   */
-  key?: string;
-
-  /**
-   * the value of the cookie (default "")
-   */
-  value?: string;
-
-  /**
-   * if set, the Expires= attribute of the cookie (defaults to the string "Infinity").
-   * See setExpires()
-   */
-  expires?: Date;
-
-  /**
-   * (seconds) if set, the Max-Age= attribute in seconds of the cookie. May also
-   * be set to strings "Infinity" and "-Infinity" for non-expiry and immediate-expiry,
-   * respectively. See setMaxAge()
-   */
-  maxAge?: number;
-
-  /**
-   * the Domain= attribute of the cookie
-   */
-  domain?: string;
-
-  /**
-   * the Path= of the cookie
-   */
-  path?: string;
-
-  /**
-   * the Secure cookie flag
-   */
-  secure?: boolean;
-
   /**
-   * the HttpOnly cookie flag
+   * Retrieve a cookie with the given domain, path and key (a.k.a. name).
+   * The RFC maintains that exactly one of these cookies should exist in a store. If the store is using versioning,
+   * this means that the latest/newest such cookie should be returned.
+   *
+   * Callback takes an error and the resulting Cookie object.
+   * If no cookie is found then null MUST be passed instead (i.e. not an error).
    */
-  httpOnly?: boolean;
+  findCookie(domain: string, path: string, key: string, cb: (error: Error, cookie: Cookie) => any): any;
```

Comment:
Yes, I think so. I used to write `any` over `void` because I had run into some random issues in the past, but the correct type should have been `void` from the beginning. Feel free to submit another PR to correct that, in the callback positions I haven't really cared since whatever the user puts is good, but it should still prefer `void` I believe.